actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -336,5 +351,18 @@ def form_authenticity_param
       def protect_against_forgery?
         allow_forgery_protection
       end
+
+      # Checks if the request originated from the same origin by looking at the
+      # Origin header.
+      def valid_request_origin?
+        return true unless forgery_protection_origin_check
+
+        origin = request.headers['origin']
+
+        # Some user agents don't send the Origin header.
+        return true if origin.blank?
```

Comment:
Would also prefer standard `if` over early return in this branch.

Another thing, isn't it only `Some older` user agents?